### PR TITLE
526 opt in email

### PR DESCRIPTION
**Pull request by SMLuthi**
## Background
We need to create a (temporary) database table to gather emails for veterans that will want to use form 526 for increase. This includes an endpoint that will create (or update) the veterans information and store it.

#### Unique to this PR

- [x] Create an endpoint to create/update the email data
- [x] Create a database migration
- [x] Create a serializer/route

#### Applies to all PRs

- [x] Appropriate test coverage & logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)

### CODE REVIEW

File: db/migrate/20180727213418_add_index_to_form526_opt_in.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class AddIndexToForm526OptIn < ActiveRecord::Migration
+  disable_ddl_transaction!
```
Comment (egarreau) [inline db/migrate/20180727213418_add_index_to_form526_opt_in.rb]: Out of curiosity, why is this necessary?

File: db/migrate/20180727211805_create_form526_opt_in.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateForm526OptIn < ActiveRecord::Migration
+  def change
+    create_table :form526_opt_ins do |t|
+      t.string :user_uuid, null: false
+      t.string :encrypted_email, null: false
+      t.string :encrypted_email_iv, null: false
```
Comment (egarreau) [inline db/migrate/20180727211805_create_form526_opt_in.rb]: What is this used for?

File: db/migrate/20180727213418_add_index_to_form526_opt_in.rb (hunk 1 above)
Comment (SMLuthi) [reply db/migrate/20180727213418_add_index_to_form526_opt_in.rb]: The short answer is because im adding a concurrent index which needs to be run outside of a transaction... but rails wraps each migration inside of a transaction (by default). So it has to be disabled.

Comment (egarreau) [reply db/migrate/20180727213418_add_index_to_form526_opt_in.rb]: What's the advantage of a concurrent index?

File: db/migrate/20180727211805_create_form526_opt_in.rb (hunk 2 above)
Comment (SMLuthi) [reply db/migrate/20180727211805_create_form526_opt_in.rb]: the `_iv` is required for the initialization vector for the cypher text (the encryption).

Comment (egarreau) [reply db/migrate/20180727211805_create_form526_opt_in.rb]: Cool, thanks for clarifying!

File: db/migrate/20180727213418_add_index_to_form526_opt_in.rb (hunk 1 above)
Comment (SMLuthi) [reply db/migrate/20180727213418_add_index_to_form526_opt_in.rb]: Its probably overkill for this case but it keeps the table from being locked out (It can still be interacted with). So it allows for zero downtime deployments.

Comment (egarreau) [reply db/migrate/20180727213418_add_index_to_form526_opt_in.rb]: I see, cool! I don't think it's overkill since it doesn't add much complexity and it seems like a nice benefit. Thanks for enlightening me :)
